server.go

```diff
@@ -511,17 +565,25 @@ func (srv *Server) serveTCP(l net.Listener) error {
 	}
 
 	for {
-		rw, err := l.Accept()
 		srv.lock.RLock()
 		if !srv.started {
 			srv.lock.RUnlock()
 			return nil
 		}
 		srv.lock.RUnlock()
+		rw, err := l.Accept()
 		if err != nil {
 			if neterr, ok := err.(net.Error); ok && neterr.Temporary() {
 				continue
 			}
+			// if listener is closed while Accept, we'll get an error
+			// we just not want to notify this error if it is related to shutdown
+			srv.lock.RLock()
```

Comment:
more locks? And again checking if we are started?

Reply:
...and then I can remove this lock.